starfish/pipeline/algorithmbase.py

```diff
@@ -1,17 +1,37 @@
+from typing import Set, Type
+
 from starfish.imagestack.imagestack import ImageStack
 from starfish.intensity_table.intensity_table import IntensityTable
 from starfish.types import LOG
 from starfish.types._constants import STARFISH_EXTRAS_KEY
 from starfish.util.logging import LogEncoder
+from .pipelinecomponent import PipelineComponent
 
 
-class AlgorithmBaseType(type):
+DISALLOWED_BASES: Set[Type] = set()
```

Comment:
what's an example of a disallowed base?

Reply:
Let me improve the documentation.  Basically, we want the subclasses of `FilterAlgorithmBase`, `RegistrationAlgorithmBase`, etc., and not the subclasses of `AlgorithmBase`.

Reply:
I went with a different approach that removed the need for this.